Test/AwsCommonRuntimeKitTests/http/HTTP2ClientConnectionTests.swift

```diff
@@ -0,0 +1,184 @@
+                actualSha.encodeToHexString().uppercased(),
+                "C7FDB5314B9742467B16BD5EA2F8012190B5E2C44A005F7984F89AAB58219534")
+    }
+
+    func testHTTP2DownloadWithHTTP1_1Request() async throws {
+        let connectionManager = try await getHttpConnectionManager(endpoint: "d1cz66xoahf9cl.cloudfront.net", alpnList: ["h2","http/1.1"])
+        let response = try await sendHTTPRequest(
+                method: "GET",
+                endpoint: "d1cz66xoahf9cl.cloudfront.net",
+                path: "/http_test_doc.txt",
+                connectionManager: connectionManager,
+                expectedVersion: expectedVersion,
+                requestVersion: .version_1_1)
+        let actualSha = try response.body.sha256()
+        XCTAssertEqual(
+                actualSha.encodeToHexString().uppercased(),
+                "C7FDB5314B9742467B16BD5EA2F8012190B5E2C44A005F7984F89AAB58219534")
+    }
+
+    func testHTTP2StreamUpload() async throws {
+        let connectionManager = try await getHttpConnectionManager(endpoint: "nghttp2.org", alpnList: ["h2"])
+        let semaphore = DispatchSemaphore(value: 0)
+        var httpResponse = HTTPResponse()
+        var onCompleteCalled = false
+        let testBody = "testBody"
+        let http2RequestOptions = try getHTTP2RequestOptions(
+                method: "PUT",
+                path: "/httpbin/put",
+                authority: "nghttp2.org",
+                body: testBody,
+                response: &httpResponse,
+                semaphore: semaphore,
+                onComplete: { stream, error in
+                    onCompleteCalled = true
+                },
+                http2ManualDataWrites: true)
+        let connection = try await connectionManager.acquireConnection()
+        let streamBase = try connection.makeRequest(requestOptions: http2RequestOptions)
+        let stream = streamBase as! HTTP2Stream
+        try stream.activate()
+        XCTAssertFalse(onCompleteCalled)
+        let data = TEST_DOC_LINE.data(using: .utf8)!
+        for chunk in data.chunked(into: 5) {
+            try await stream.writeData(data: chunk, endOfStream: false)
+            XCTAssertFalse(onCompleteCalled)
+        }
+
+        XCTAssertFalse(onCompleteCalled)
+        // Sleep for 5 seconds to make sure onComplete is not triggerred until endOfStream is true
+        try await Task.sleep(nanoseconds: 5_000_000_000)
+        XCTAssertFalse(onCompleteCalled)
+        try await stream.writeData(data: Data(), endOfStream: true)
+        semaphore.wait()
+        XCTAssertTrue(onCompleteCalled)
+        XCTAssertNil(httpResponse.error)
+        XCTAssertEqual(httpResponse.statusCode, 200)
+
+        // Parse json body
+        struct Response: Codable {
+            let data: String
+        }
+
+        let body: Response = try! JSONDecoder().decode(Response.self, from: httpResponse.body)
+        XCTAssertEqual(body.data, testBody + TEST_DOC_LINE)
+    }
+}
```

Comment:
could you add one test case for bidi-stream?

Reply:
I am not sure what do you mean by bidi-stream. This is similar, where we send data using writeData and receive data in the onBody callback until endOfStream is true.

Reply:
I would have expected a test case where both writeData and onBody used asynchronously.

Reply:
@ganeshnj They are used both in `testHTTP2StreamUpload`. `nghttp2.org/httpbin/put` is a server which just echoes back whatever we send. So we send data in chunks to `nghttp2.org/httpbin/put` and await stream to complete. We receive data in onBody in a different thread, and onComplete is triggered. At the end we compare the data.